nannyml/plots/components/figure.py

```diff
@@ -165,7 +165,7 @@ def add_confidence_band(
             x, data = add_artificial_endpoint(indices, start_dates, end_dates, data)
 
         show_in_legend = kwargs.get('showlegend', True)
-        if subplot_args is None:
+        if subplot_args is None or (subplot_args.get('row') is None and subplot_args.get('col') is None):
```

Comment:
You might turn the statement `(subplot_args.get('row') is None and subplot_args.get('col') is None)` into its own well-named variable, that keeps the if statement concise and clear :+1: